psi4/src/psi4/libmints/local.cc

```diff
@@ -549,4 +550,544 @@ void PMLocalizer::localize() {
     U_->transpose_this();
 }
 
+IBOLocalizer::IBOLocalizer(std::shared_ptr<BasisSet> primary, std::shared_ptr<BasisSet> minao,
+                             std::shared_ptr<Matrix> C)
+    : Localizer(primary, C), minao_(minao) {
+    common_init();
+}
+IBOLocalizer::~IBOLocalizer() {}
+void IBOLocalizer::common_init() {
+    print_ = 0;
+    debug_ = 0;
+    bench_ = 0;
+    convergence_ = 1.0E-12;
+    maxiter_ = 50;
+    use_ghosts_ = false;
+    power_ = 4;
+    condition_ = 1.0E-7;
+    use_stars_ = false;
+    stars_completeness_ = 0.9;
+    stars_.clear();
+}
+std::shared_ptr<IBOLocalizer> IBOLocalizer::build(std::shared_ptr<BasisSet> primary, std::shared_ptr<BasisSet> minao,
+                                                    std::shared_ptr<Matrix> C, Options& options) {
+
+    auto local = std::make_shared<IBOLocalizer>(primary, minao, C);
+
+    local->set_print(options.get_int("PRINT"));
+    local->set_debug(options.get_int("DEBUG"));
+    local->set_bench(options.get_int("BENCH"));
+    local->set_convergence(options.get_double("LOCAL_CONVERGENCE"));
+    local->set_maxiter(options.get_int("LOCAL_MAXITER"));
+    local->set_use_ghosts(options.get_bool("LOCAL_USE_GHOSTS"));
+    local->set_condition(options.get_double("LOCAL_IBO_CONDITION"));
+    local->set_power(options.get_double("LOCAL_IBO_POWER"));
+    local->set_use_stars(options.get_bool("LOCAL_IBO_USE_STARS"));
+    local->set_stars_completeness(options.get_double("LOCAL_IBO_STARS_COMPLETENESS"));
+
+    std::vector<int> stars;
+    for (int ind = 0; ind < options["LOCAL_IBO_STARS"].size(); ind++) {
+        stars.push_back(options["LOCAL_IBO_STARS"][ind].to_integer() - 1);
+    }
+    local->set_stars(stars);
```

Comment:
Why is this not in `common_init`?

The current content of `common_init` is irrelevant when you do this setting right after.

Reply:
I am following the convention of the base Localizer class, which does the same thing (initializes instance variables in `common_init`, and then sets them with options in the `Localizer::build` method.

Reply:
For future reference, "following the convention" is often not a good argument if there's not a good reason for the convention.

But in this case, there is a reason that I missed the first time. Add a comment in the header file saying that the user will normally set options by passing in an `options` object through `build`, but options are initialized in `common_init` in case that doesn't happen.